crates/node/src/sync.rs

```diff
@@ -166,9 +190,21 @@ impl Node {
             }
         }
 
+        let identities = self.ctx_manager.get_context_owned_identities(context.id)?;
+
+        let Some(our_identity) = identities.into_iter().choose(&mut thread_rng()) else {
+            bail!("no identities found for context: {}", context.id);
+        };
```

Comment:
this already happens inside `handle_blob_share_request` and `handle_state_sync_request`